### PR TITLE
Added retry and retryWhen support for Single

### PR DESCRIPTION
**Pull request by klemzy**

### CODE REVIEW

File: src/test/java/rx/SingleTest.java [hunk 1]
```diff
@@ -1,45 +1,18 @@
 /**
  * Copyright 2015 Netflix, Inc.
- *
+ * <p/>
```
Comment (akarnokd) [inline src/test/java/rx/SingleTest.java]: What happened to the header?

**Comment by akarnokd** [comment]
Please don't unroll `import *` because it creates merge conflicts all the time.

**Comment by klemzy** [comment]
My codestyle settings messed everything a bit. Is there a exported codestyle.xml file or something similar so I could import into my IDE?

**Comment by akarnokd** [comment]
Not really. Also please squash your commits.

File: src/main/java/rx/Single.java [hunk 2]
```diff
@@ -2209,4 +2196,101 @@ public void call(SingleSubscriber<? super T> singleSubscriber) {
...
+         return toObservable().retry().toSingle();
+     }
+
+     /**
+      * Returns an Single that mirrors the source Single, resubscribing to it if it calls {@code onError}
+      * up to a specified number of retries.
+      *
+      * <img width="640" height="315" src="https://raw.github.com/wiki/ReactiveX/RxJava/images/rx-operators/retry.png" alt="">
+      *
+      * If the source Single calls {@link Observer#onError}, this method will resubscribe to the source
```
Comment (akarnokd) [inline src/main/java/rx/Single.java]: SingleSubscriber#onError

File: src/main/java/rx/Single.java [hunk 3]
```diff
@@ -2209,4 +2196,101 @@ public void call(SingleSubscriber<? super T> singleSubscriber) {
...
         return singlesArray;
     }
+
+    /**
+      * Returns a Single that mirrors the source Single, resubscribing to it if it calls {@code onError}
+      * (infinite retry count).
+      *
+      * <img width="640" height="315" src="https://raw.github.com/wiki/ReactiveX/RxJava/images/rx-operators/retry.png" alt="">
+      *
+      * If the source Single calls {@link Observer#onError}, this method will resubscribe to the source
```
Comment (akarnokd) [inline src/main/java/rx/Single.java]: SingleSubscriber#onError

**Comment by akarnokd** [comment]
Could you also fix those 1 spaces removed everywhere?

**Comment by klemzy** [comment]
I have fixed all reported problems.

File: src/main/java/rx/Single.java [hunk 4]
```diff
@@ -1,50 +1,37 @@
 /**
  * Copyright 2015 Netflix, Inc.
- * 
+ *
```
Comment (akarnokd) [inline src/main/java/rx/Single.java]: These space removals are still there. Are you using IntelliJ? I'm sure there is a way to disable the remove whitespace from line ends option.

Comment (klemzy) [inline src/main/java/rx/Single.java]: Yes I am using Intellij. Ok I will check.

**Comment by klemzy** [comment]
Something went really wrong. I merged latest changes from upstream 1.x to my 1.x branch.

**Comment by akarnokd** [comment]
In this case, I'd just start from a clean branch of master and reapply/merge my customizations.

**Comment by klemzy** [comment]
I managed to fix it. Now I just need to sort out the missing spaces.

**Comment by akarnokd** [comment]
This looks much better.

:+1:

**Comment by stevegury** [comment]
:+1:

**Pull request merged**